### PR TITLE
Fixes usage of Code component in Vercel

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Fixes https://github.com/withastro/astro/issues/6174

## Testing

- Created some unit tests for this, so we can better test how we generate options in the future.

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
!preview shiki-vercel

**Comment by matthewp** [comment]
Working: https://astro-vercel-shiki-error-kljl7rwie-matthew6.vercel.app/

File: packages/astro/components/Shiki.js [hunk 1]
```diff
@@ -15,14 +15,14 @@ function stringify(opts) {
 
 /**
  * @param {import('shiki').HighlighterOptions} opts
- * @returns {Promise<import('shiki').Highlighter>}
+ * @returns {Promise<import('shiki').HighlighterOptions>}
  */
-async function resolveHighlighter(opts) {
+export async function resolveHighlighterOptions(opts) {
```
Comment (natemoo-re) [inline packages/astro/components/Shiki.js:20]: So the `export` here is the fix?

**Review by natemoo-re — COMMENTED** [review]
Code LGTM, but would love to hear more about how this fix works

File: packages/astro/components/Shiki.js (hunk 1 above)
Comment (matthewp) [reply packages/astro/components/Shiki.js:20]: No, the order swapping is the fix: https://github.com/withastro/astro/pull/6198/files#diff-766f75438aeb2486b5bf2c8f6f3a6521fb48e42b6b51b9be6077cdad5f85c162R22-R25

the export is to unit test this function.

Comment (natemoo-re) [reply packages/astro/components/Shiki.js:20]: Ahhh OK thanks!

**Review by natemoo-re — APPROVED** [review]

**Pull request merged**